packages/component-base/src/polylit-mixin.js

```diff
@@ -0,0 +1,267 @@
+
+      let result = defaultDescriptor;
+
+      if ('value' in options) {
+        // Set the default value
+        this.addInitializer((instance) => {
+          if (typeof options.value === 'function') {
+            instance[name] = options.value.call(instance);
+          } else {
+            instance[name] = options.value;
+          }
+        });
+      }
+
+      if (options.readOnly) {
+        const setter = defaultDescriptor.set;
+
+        this.addInitializer((instance) => {
+          // This is run during construction of the element
+          instance['_set' + upper(name)] = function (value) {
+            setter.call(instance, value);
+          };
+        });
+
+        result = {
+          get: defaultDescriptor.get,
+          set() {
+            // Do nothing, property is read-only.
+          },
+          configurable: true,
+          enumerable: true
+        };
+      }
+
+      if (options.observer) {
+        const method = options.observer;
+
+        // set this method
+        this.getOrCreateMap('__observers').set(name, method);
+
+        this.addInitializer((instance) => {
+          if (!instance[method]) {
+            console.warn(`observer method ${method} not defined`);
+          }
+        });
+      }
+
+      if (options.notify) {
+        if (!this.__notifyProps) {
+          this.__notifyProps = new Set();
+          // eslint-disable-next-line no-prototype-builtins
+        } else if (!this.hasOwnProperty('__notifyProps')) {
+          // clone any existing observers (superclasses)
+          const notifyProps = this.__notifyProps;
+          this.__notifyProps = new Set(notifyProps);
+        }
+
+        // set this method
+        this.__notifyProps.add(name);
+      }
+
+      if (options.computed) {
+        const assignComputedMethod = `__assignComputed${name}`;
+        const observer = parseObserver(options.computed);
+        this.prototype[assignComputedMethod] = function (...props) {
+          this[name] = this[observer.method](...props);
+        };
+
+        this.getOrCreateMap('__complexObservers').set(assignComputedMethod, observer.observerProps);
+      }
+
+      if (!options.attribute) {
+        options.attribute = name.replace(/([A-Z])/g, '-$1').toLowerCase();
+      }
+
+      return result;
+    }
+
+    /** @protected */
+    ready() {
+      if (super.ready) {
+        super.ready();
+      }
+      this.$ = this.$ || {};
+      this.shadowRoot.querySelectorAll('[id]').forEach((node) => {
+        this.$[node.id] = node;
+      });
+    }
+
+    /** @protected */
+    firstUpdated() {
+      super.firstUpdated();
+
+      this.ready();
+    }
+
+    /** @protected */
+    // willUpdate(props) {
+    updated(props) {
+      if (this.constructor.__observers) {
+        this.__runObservers(props, this.constructor.__observers);
+      }
+
+      if (this.constructor.__notifyProps) {
+        this.__runNotifyProps(props, this.constructor.__notifyProps);
```

Comment:
Looks like Polymer runs [notify effects after observers](https://github.com/Polymer/polymer/blob/65039eebe167bf110850ea0320f87bb3014c5c24/lib/mixins/property-effects.js#L1921-L1926) - maybe we need to change the order so that `PolylitMixin` also runs them after complex and dynamic observers are executed.